config/build/rollup.config.js

```diff
@@ -0,0 +1,324 @@
+            // We don't want to do process.env.NODE_ENV checks in our main
+            // builds. Our consumers should handle that. However, if we
+            // do our prod build, we do want to do this.
+            replace({
+                preventAssignment: true,
+                values: valueReplacementMappings,
+            }),
+            babel({
+                babelHelpers: "bundled",
+                presets: createBabelPresets({platform, format}),
+                plugins: createBabelPlugins({platform, format}),
+                exclude: "node_modules/**",
+            }),
+            resolve({
+                browser: platform === "browser",
+            }),
+            autoExternal({
+                packagePath: makePackageBasedPath(name, "./package.json"),
+            }),
+            terser(),
+            ...plugins,
+        ],
+    };
+
+    return config;
+};
+
+/**
+ * Get the configurations for building a package.
+ *
+ * For each package in our packages folder, generate the outputs we want.
+ *
+ * To determine what those outputs are, we read the `package.json` file for
+ * each package. If the package has a `browser` field, then we generate
+ * browser and node assets. If not, we just generate the node assets.
+ * Note that we also get the output paths from the package.json.
+ *
+ * We also can filter the outputs based on command line options:
+ * `--configPlatforms` - Comma-separated list. Valid values are "browser"
+ *                       and "node".
+ * `--configFormats`   - Comma-separated list. Valid values are "cjs" and
+ *                       "esm". If not specified, then we generate both.
+ */
+const getPackageInfo = (commandLineArgs, pkgName) => {
+    const pkgJsonPath = makePackageBasedPath(pkgName, "./package.json");
+    if (!fs.existsSync(pkgJsonPath)) {
+        return [];
+    }
+    const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath));
+
+    // Determine what formats and platforms we are building.
+    const platforms = getPlatforms(commandLineArgs);
+    const formats = getFormats(commandLineArgs);
+
+    // Now we have the package.json, we need to look at the main, module, and
+    // browser fields and values.
+    const {main: cjsNode, module: esmNode, browser} = pkgJson;
+    const cjsBrowser = browser == null ? null : browser[cjsNode];
+    const esmBrowser = browser == null ? null : browser[esmNode];
+
+    // This generates the flow import file and a file for intellisense to work.
+    // By using the same instance of it across all output configurations
+    // while using the `copyOnce` value, we ensure it only gets copied one time.
+    const typesAndDocsCopy = copy({
+        copyOnce: true,
+        verbose: true,
+        targets: [
+            // This is the file that provides flow types.
+            {
+                // src path is relative to the package root unless started
+                // with ./
+                src: "build-settings/index.js.flow.template",
```

Comment:
I just realized I forgot to include this file.  I'll do that as part of this PR.